redhat-appstudio/remote-secret
# Remote secret rejected data upload metric

### What does this PR do?

PR adds metrics for the failed upload data operations, with the operation name and failure reason labels, so the counterscan be filtered/ordered by them.

### Screenshot/screencast of this PR

![image](https://github.com/redhat-appstudio/remote-secret/assets/1651062/365bd388-53c6-4e4f-9309-3f5be909b98f)


### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-575

### How to test this PR?

1) Need to setup minikube with moniroting (PR is on the way)
2) Try to upload data with some error condidion  (like duplicate field, or bad source secret etc)
(can use the `remote-secret-conflicting-targets` sample or example below )

```
apiVersion: appstudio.redhat.com/v1beta1
kind: RemoteSecret
metadata:
  name: myrs1
  namespace: target-namespace
spec:
  secret:
    generateName: copied-
dataFrom:
  name: non-existent-secret
  namespace: default
```


3) Verify counter incremented with correct labels
